include/deal.II/lac/precondition.h

```diff
@@ -2219,24 +2248,21 @@ PreconditionChebyshev<MatrixType, VectorType, PreconditionerType>::
     PreconditionChebyshev<MatrixType, VectorType, PreconditionerType> *>(this)
     ->theta = (max_eigenvalue + alpha) * 0.5;
 
-  // We do not need the third auxiliary vector in case we have a
+  // We do not need the second temporary vector in case we have a
   // DiagonalMatrix as preconditioner and use deal.II's own vectors
+  using NumberType = typename VectorType::value_type;
   if (std::is_same<PreconditionerType, DiagonalMatrix<VectorType>>::value ==
         false ||
-      (std::is_same<VectorType,
-                    dealii::Vector<typename VectorType::value_type>>::value ==
-         false &&
-       ((std::is_same<
-           VectorType,
-           LinearAlgebra::distributed::Vector<typename VectorType::value_type,
-                                              MemorySpace::Host>>::value ==
+      (std::is_same<VectorType, dealii::Vector<NumberType>>::value == false &&
+       ((std::is_same<VectorType,
+                      LinearAlgebra::distributed::
+                        Vector<NumberType, MemorySpace::Host>>::value ==
          false) ||
-        (std::is_same<
-           VectorType,
-           LinearAlgebra::distributed::Vector<typename VectorType::value_type,
-                                              MemorySpace::CUDA>>::value ==
+        (std::is_same<VectorType,
+                      LinearAlgebra::distributed::
+                        Vector<NumberType, MemorySpace::CUDA>>::value ==
          false))))
-    update3.reinit(src, true);
+    temp_vector2.reinit(src, true);
```

Comment:
does it make sense to make this invalid/size 0 or something in the else branch (at least in debug mode)?

Reply:
We should make this size zero, yes, just to be sure, even though there is probably nothing there because it never should get initialized elsewhere for the given template arguments.